docs/source/changelog.md

```diff
@@ -14,6 +14,10 @@ this list should be updated.
 
 - K8s 1.22 is now required.
 - The Helm chart's provided images now use Python 3.11 instead of Python 3.9.
+- JupyterHub 3.0.0 is upgraded to 4.0.0b1.
+  - The upgrade does not require restarts of single-user servers', and they can
+    have either jupyterhub 3 or 4 installed both before and after the upgrade.
```

Comment:
```suggestion
  - Although not officially supported running JupyterHub 4.0.0 with jupyterhub-singleuser 3.* servers should work,
    and should not require restarts of single-user servers.
    However we recommend your singleuser images are upgraded as soon as practical.
```
I think it's better to keep expectations low- in the event of future unforeseen problems it gives us more leeway to break compatibility between major versions.

Reply:
Thank you @manics for reviewing! Good point!

I wanted to capure these two separte aspects quite explicitly, so I iterated on the suggestion you provided
- we expect (but don't guarantee) that user servers can keep running during this major version upgrade
- we expect (but don't guarantee) that user servers can use jupyterhub-singleuser 3 or 4, against the jupyterhub server version 3 (before chart upgrade) or 4 (after chart upgrade).

```suggestion
  - Although it is not officially supported to run a JupyterHub server with a
    major version different from the singleuser servers' `jupyterhub-singleuser`
    version, it seems possible during this upgrade. We recommend your singleuser
    images are upgraded as soon as practical, before or after the JupyterHub
    chart is upgraded. We expect running user servers to be able to keep running
    during the upgrade.
```

What do you think?

Reply:
Sounds fine to me!